src/App.tsx

```diff
@@ -1,14 +1,72 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { filterTodos } from './helpers';
+import { FilterOptions } from './types/FilterOptions';
+import { DEFAULT_FILTER } from './constants';
 
 export const App: React.FC = () => {
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [isTodosLoading, setIsTodosLoading] = useState(false);
+  const [filterOptions, setFilterOptions] = useState<FilterOptions>(
+    DEFAULT_FILTER,
+  );
+
+  useEffect(() => {
+    setIsTodosLoading(true);
+    getTodos()
+      .then(setVisibleTodos)
+      .finally(() => setIsTodosLoading(false));
+  }, []);
+
+  useEffect(() => {
+    getTodos().then((todos) => {
```

Comment:
Why do you need to fetch todos again? You already have them, you need to refactor your solution because it's a bad idea to make API calls if it is not really needed
You can just create variable `const filteredTodos = filterTodos(visibleTodos, filterOptions);`
And than just render todos from this constant
P.S. But in this case you need to rename state(it's not a visibleTodos anymore)